system/default_ad_api/src/compatibility/autoware_state.cpp

```diff
@@ -15,8 +15,86 @@
+    return 3.0;
+  }
+  if (state == State::ARRIVED_GOAL) {
+    return 2.0;
+  }
+  return 0.0;
+}
+
+const double arrived_goal_timeout = get_wait_time(State::ARRIVED_GOAL);
+
+// clang-format off
+std::unordered_map<std::pair<Value, Value>, Value, PairHash> matrix =
+{
+  {{State::INITIALIZING      , State::INITIALIZING      }, State::INITIALIZING      },
+  {{State::INITIALIZING      , State::WAITING_FOR_ROUTE }, State::WAITING_FOR_ROUTE },
+  {{State::INITIALIZING      , State::PLANNING          }, State::WAITING_FOR_ROUTE },
+  {{State::INITIALIZING      , State::WAITING_FOR_ENGAGE}, State::WAITING_FOR_ROUTE },
+  {{State::INITIALIZING      , State::DRIVING           }, State::WAITING_FOR_ROUTE },
+  {{State::INITIALIZING      , State::ARRIVED_GOAL      }, State::WAITING_FOR_ROUTE },
+  {{State::WAITING_FOR_ROUTE , State::INITIALIZING      }, State::WAITING_FOR_ROUTE },
+  {{State::WAITING_FOR_ROUTE , State::WAITING_FOR_ROUTE }, State::WAITING_FOR_ROUTE },
+  {{State::WAITING_FOR_ROUTE , State::PLANNING          }, State::PLANNING          },
+  {{State::WAITING_FOR_ROUTE , State::WAITING_FOR_ENGAGE}, State::PLANNING          },
+  {{State::WAITING_FOR_ROUTE , State::DRIVING           }, State::DRIVING           },
+  {{State::WAITING_FOR_ROUTE , State::ARRIVED_GOAL      }, State::DRIVING           },
+  {{State::PLANNING          , State::INITIALIZING      }, State::PLANNING          },
+  {{State::PLANNING          , State::WAITING_FOR_ROUTE }, State::WAITING_FOR_ROUTE },
+  {{State::PLANNING          , State::PLANNING          }, State::PLANNING          },
+  {{State::PLANNING          , State::WAITING_FOR_ENGAGE}, State::WAITING_FOR_ENGAGE},
+  {{State::PLANNING          , State::DRIVING           }, State::WAITING_FOR_ENGAGE},
+  {{State::PLANNING          , State::ARRIVED_GOAL      }, State::WAITING_FOR_ENGAGE},
+  {{State::WAITING_FOR_ENGAGE, State::INITIALIZING      }, State::WAITING_FOR_ENGAGE},
+  {{State::WAITING_FOR_ENGAGE, State::WAITING_FOR_ROUTE }, State::WAITING_FOR_ROUTE },
+  {{State::WAITING_FOR_ENGAGE, State::PLANNING          }, State::PLANNING          },
+  {{State::WAITING_FOR_ENGAGE, State::WAITING_FOR_ENGAGE}, State::WAITING_FOR_ENGAGE},
+  {{State::WAITING_FOR_ENGAGE, State::DRIVING           }, State::DRIVING           },
+  {{State::WAITING_FOR_ENGAGE, State::ARRIVED_GOAL      }, State::ARRIVED_GOAL      },
+  {{State::DRIVING           , State::INITIALIZING      }, State::DRIVING           },
+  {{State::DRIVING           , State::WAITING_FOR_ROUTE }, State::WAITING_FOR_ROUTE },
+  {{State::DRIVING           , State::PLANNING          }, State::PLANNING          },
+  {{State::DRIVING           , State::WAITING_FOR_ENGAGE}, State::WAITING_FOR_ENGAGE},
+  {{State::DRIVING           , State::DRIVING           }, State::DRIVING           },
+  {{State::DRIVING           , State::ARRIVED_GOAL      }, State::ARRIVED_GOAL      },
+  {{State::ARRIVED_GOAL      , State::INITIALIZING      }, State::DRIVING           },
```

Comment:
Seeing other states, is here a mistake?

```suggestion
  {{State::ARRIVED_GOAL      , State::INITIALIZING      }, State::ARRIVED_GOAL      },
```